File: Ejemplos/Sincronismo/C++/NoAtomic.cpp
#include <iostream>
#include <thread>
#include <vector>

int count=0;

void Sum()
{
    for (int i=0; i<100; ++i) 
    {
        count++;
    }
}

int main() 
{
    std::vector<std::thread> threads;

    for(int i=0; i<1000; ++i)
    {
        threads.push_back( std::thread(Sum) );
    }

    for(int i=0; i<1000; ++i)
    {
        threads[i].join();
    }

    std::cout<<"Resultado "<<count<<std::endl;

    return EXIT_SUCCESS;
} 

File: Ejemplos/Sincronismo/Java/NoAtomic.java
public class NoAtomic 
{
    static int count = 0;

    static class MyThread extends Thread 
   {
        public void run()
        {
            for (int i=0; i<100; ++i) 
            {
                count++;
            }
        }
    }

    public static void main(String[] args) throws InterruptedException 
    {
        MyThread []threads = new MyThread[1000];
       
        for(int i=0; i<1000; ++i)
        {
            threads[i] = new MyThread();
        }

        for(int i=0; i<1000; ++i)
        {
            threads[i].start();
        }

        for(int i=0; i<1000; ++i)
        {
            threads[i].join();
        }

        System.out.println("Valor: " + count);		
    }
}
